--- Makefile.orig	2009-11-27 12:04:33.000000000 -0800
+++ Makefile	2009-11-27 12:11:47.786739617 -0800
@@ -25,12 +25,22 @@
 LIBS	= -L. -llist
 #LIBS	= -L . -llist   # Use this for HP-UX; great loader guys!
 CPP	= cpp -E -P -C
+CC	= cc
+LD	= cc
 MANDIR	= /usr/share/man
 LIBDIR	= /usr/lib
 INCDIR	= /usr/include
 
 #add macro for ranlib 4/96 *kob* - ranlib doesn't exist on solaris
 RANLIB = ls
+LN = ln -s
+
+# try making shared lib (SLA 11/2009)
+SHARED = liblist.so
+LDFLAGS += -L$(LIBDIR)
+MKOBJ = $(CC) -fPIC -c list.c -I$(INCDIR)
+MKSO = -shared -Wl,-soname,$(SHARED)
+SOVER = 0.0
 
 # We specify some goofy dependencies between the man pages and the source,
 # because the man page reflects whether USE_MACROS was specified.  Thus,
@@ -42,8 +52,16 @@
 		(cd examples; make)
 
 liblist.a:	list.o
+		@echo "Making static lib ..." 
 		ar rc liblist.a list.o
 		$(RANLIB) liblist.a
+		@echo "done"
+		@echo "Making shared lib $(SHARED) ..."
+		$(MKOBJ) -I.
+		$(LD) -I. $(MKSO) list.o -o $(SHARED).$(SOVER) $(LDFLAGS)
+		$(LN) $(SHARED).$(SOVER) $(SHARED).0
+		$(LN) $(SHARED).$(SOVER) $(SHARED)
+		@echo "done"
 
 list.o:		list.h list.3 Makefile
 
@@ -62,6 +80,7 @@
 install:
 		install -c liblist.a $(DESTDIR)$(LIBDIR)
 		$(RANLIB) $(DESTDIR)$(LIBDIR)/liblist.a
+		install -c liblist.so $(DESTDIR)$(LIBDIR)
 		install -c list.h $(DESTDIR)$(INCDIR)
 		install -c queue.h $(DESTDIR)$(INCDIR)
 		install -c stack.h $(DESTDIR)$(INCDIR)
@@ -73,5 +92,5 @@
 		install -c stack.3 $(DESTDIR)$(MANDIR)/man3
 
 clean:
-		rm -f *.o *.a list.h *.3 core
+		rm -f *.o *.a *.so list.h *.3 core
 		(cd examples; make clean)
