From 5ef9fd977889a3029d2ab0d44efecb2e841c6a72 Mon Sep 17 00:00:00 2001
From: Stephen Arnold <nerdboy@gentoo.org>
Date: Sun, 24 Nov 2019 18:55:12 -0800
Subject: [PATCH] nanomsg/__init__.py: fix ctype int for _INT_PACKER

Signed-off-by: Stephen Arnold <nerdboy@gentoo.org>
---
 nanomsg/__init__.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/nanomsg/__init__.py b/nanomsg/__init__.py
index ab584dc..93eb3f7 100644
--- a/nanomsg/__init__.py
+++ b/nanomsg/__init__.py
@@ -159,7 +159,7 @@ class Socket(object):
 
     """
 
-    _INT_PACKER = _Struct(str('l'))
+    _INT_PACKER = _Struct(str('i'))
 
     class _Endpoint(object):
         def __init__(self, socket, endpoint_id, address):
-- 
2.21.0
