diff -u old/pwin32.h new/pwin32.h
--- old/pwin32.h	2011-02-13 18:24:32.000000000 +0300
+++ new/pwin32.h	2011-02-13 18:41:25.956000006 +0300
@@ -147,15 +147,13 @@
 typedef unsigned int uint32_t;
 #endif
 
-#if ! U_HAVE_INT64_T
+#ifndef __int8_t_defined
 #if U_INT64_IS_LONG_LONG
     typedef signed long long int64_t;
 #else
     typedef signed __int64 int64_t;
 #endif
-#endif
 
-#if ! U_HAVE_UINT64_T
 #if U_INT64_IS_LONG_LONG
     typedef unsigned long long uint64_t;
 #else
