--- ScientificPython-2.9.0.orig/setup.py	2008-10-28 14:52:30.000000000 +0100
+++ ScientificPython-2.9.0/setup.py	2010-02-02 14:58:55.029160416 +0100
@@ -3,6 +3,7 @@
 from distutils.core import setup, Extension
 from distutils.command.install_headers import install_headers
 import os, sys
+from subprocess import Popen, PIPE
 from glob import glob
 
 class Dummy:
@@ -106,6 +107,8 @@
     else:
         print "Using netCDF installation in ", netcdf_prefix
         netcdf_include = os.path.join(netcdf_prefix, 'include')
+        mpi_ccflags = Popen(["mpicc", "-showme:compile"], stdout=PIPE).communicate()[0].rstrip().split()
+        mpi_ldflags = Popen(["mpicc", "-showme:link"], stdout=PIPE).communicate()[0].rstrip().split()
         netcdf_h_file = os.path.join(netcdf_prefix, 'include', 'netcdf.h')
         netcdf_lib = os.path.join(netcdf_prefix, 'lib')
     ext_modules = [Extension('Scientific_netcdf',
@@ -114,7 +117,7 @@
                                           + arrayobject_h_include,
                              library_dirs=[netcdf_lib],
                              libraries = ['netcdf'],
-                             extra_compile_args=extra_compile_args)]
+                             extra_compile_args=extra_compile_args + mpi_ccflags +mpi_ldflags )]
 
 try:
     # Add code for including documentation in Mac packages
