diff --git a/etc/fpnd.ini b/etc/fpnd.ini
index 7c9012b..cb44392 100644
--- a/etc/fpnd.ini
+++ b/etc/fpnd.ini
@@ -5,6 +5,6 @@ log_name = fpnd.log
 pid_name = fpnd.pid
 
 [Paths]
-home_dir = /usr/lib/fpnd/
+home_dir = /usr/libexec/fpnd/
 log_path = /var/log/
 pid_path = /run/
diff --git a/etc/fpnd.service b/etc/fpnd.service
index c7a229f..bf9df09 100644
--- a/etc/fpnd.service
+++ b/etc/fpnd.service
@@ -4,8 +4,8 @@ Requires=zerotier-one.service
 After=zerotier-one.service
 
 [Service]
-ExecStart=/usr/lib/fpnd/fpnd.py start
-ExecStop=/usr/lib/fpnd/fpnd.py stop
+ExecStart=/usr/sbin/fpnd.py start
+ExecStop=/usr/sbin/fpnd.py stop
 PIDFile=/run/fpnd.pid
 Type=simple
 Restart=on-failure
diff --git a/setup.py b/setup.py
index 8e444f1..8b2bed7 100644
--- a/setup.py
+++ b/setup.py
@@ -27,14 +27,16 @@ setup(
     name='fpnd',
     packages=['node_tools',],
     data_files=[
-        ('lib/fpnd', ['bin/fpn0-down.sh',
-                      'bin/fpn0-setup.sh',
-                      'bin/fpn1-down.sh',
-                      'bin/fpn1-setup.sh',
-                      'etc/fpnd.ini',
-                      'scripts/fpnd.py',
-                      'scripts/msg_responder.py',
-                      'scripts/msg_subscriber.py']),
+        ('libexec/fpnd', ['bin/fpn0-down.sh',
+                          'bin/fpn0-setup.sh',
+                          'bin/fpn1-down.sh',
+                          'bin/fpn1-setup.sh',
+                          'etc/fpnd.ini',
+                          'scripts/msg_responder.py',
+                          'scripts/msg_subscriber.py']),
+    ],
+    scripts=[
+        'scripts/fpnd.py'
     ],
     version=__version__,
     license='AGPL-3.0',
