diff --git a/setup.py b/setup.py
index 693e246..801f7a9 100644
--- a/setup.py
+++ b/setup.py
@@ -48,13 +48,6 @@ setup(
                 'writing lightweight networked services using nanomsg',
     long_description=read_long_description('README.md'),
     download_url=NANOSERVICE_DOWNLOAD_URL,
-    install_requires=[
-        'msgpack',
-        'nanomsg @ git+https://github.com/freepn/nanomsg-python@master',
-    ],
-    dependency_links=[
-        'git+https://github.com/freepn/nanomsg-python.git@master#egg=nanomsg',
-    ],
     classifiers=[
         "Development Status :: 4 - Beta",
         "Intended Audience :: Developers",
