diff --git a/etc/fpnd.ini b/etc/fpnd.ini
index 7c9012b..6e08a6c 100644
--- a/etc/fpnd.ini
+++ b/etc/fpnd.ini
@@ -5,6 +5,6 @@ log_name = fpnd.log
 pid_name = fpnd.pid
 
 [Paths]
-home_dir = /usr/lib/fpnd/
-log_path = /var/log/
+home_dir = /usr/libexec/fpnd/
+log_path = /var/log/fpnd/
 pid_path = /run/
diff --git a/etc/fpnd.openrc b/etc/fpnd.openrc
index 2fa864f..97050e7 100755
--- a/etc/fpnd.openrc
+++ b/etc/fpnd.openrc
@@ -3,7 +3,7 @@
 # Distributed under the terms of the GNU General Public License v2
 
 description="fpnd - FreePN daemon"
-command="/usr/bin/fpnd.py"
+command="/usr/sbin/fpnd.py"
 pidfile="/var/run/${SVCNAME}.pid"
 
 depend() {
diff --git a/setup.py b/setup.py
index 593f990..f296e70 100644
--- a/setup.py
+++ b/setup.py
@@ -27,14 +27,14 @@ setup(
     name='fpnd',
     packages=['node_tools',],
     data_files=[
-        ('lib/fpnd', ['bin/fpn0-down.sh',
-                      'bin/fpn0-setup.sh',
-                      'bin/fpn1-down.sh',
-                      'bin/fpn1-setup.sh',
-                      'etc/fpnd.ini',
-                      'scripts/fpnd.py',
-                      'scripts/fpn_cache_check.py']),
-        ('bin', ['bin/fpnd']),
+        ('libexec/fpnd', ['bin/fpn0-down.sh',
+                          'bin/fpn0-setup.sh',
+                          'bin/fpn1-down.sh',
+                          'bin/fpn1-setup.sh',
+                          'scripts/fpn_cache_check.py']),
+    ],
+    scripts=[
+        'scripts/fpnd.py'
     ],
     version=__version__,
     license='AGPL-3.0',
