diff --git a/src/robot/running/timeouts/ironpython.py b/src/robot/running/timeouts/ironpython.py
index 91155c5b5..6d03a395c 100644
--- a/src/robot/running/timeouts/ironpython.py
+++ b/src/robot/running/timeouts/ironpython.py
@@ -54,4 +54,4 @@ class Runner(object):
     def get_result(self):
         if not self._error:
             return self._result
-        raise self._error[0], self._error[1], self._error[2]
+	raise self._error[0](self._error[1]).with_traceback(self._error[2])
