From c881fbf455757f7cad8ee2b25574e43baa0e5037 Mon Sep 17 00:00:00 2001
From: Stephen Arnold <nerdboy@gentoo.org>
Date: Sun, 2 Dec 2018 11:25:32 -0800
Subject: [PATCH 3/3] setup.py: small patch to enable USE=jython and install
 jybot scripts

Signed-off-by: Stephen Arnold <nerdboy@gentoo.org>
---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 7576bd097..d28f8a6fc 100755
--- a/setup.py
+++ b/setup.py
@@ -59,7 +59,7 @@ if sys.platform.startswith('java'):
 elif sys.platform == 'cli':
     SCRIPTS = ['ipybot', 'ipyrebot']
 else:
-    SCRIPTS = ['pybot', 'rebot']
+    SCRIPTS = ['pybot', 'rebot', 'jybot', 'jyrebot']
 SCRIPTS = [join('src', 'bin', s) for s in SCRIPTS]
 if os.sep == '\\':
     SCRIPTS = [s+'.bat' for s in SCRIPTS]
-- 
2.18.0

