--- a/NagAconda/Plugin.py	2018-06-19 15:58:05.845893193 -0700
+++ b/NagAconda/Plugin.py	2018-06-19 15:58:55.314106832 -0700
@@ -444,11 +444,11 @@
         # We'll use the opportunity to check the status ranges right when the
         # variable is set so we don't have to loop through all of them later.
 
-        if len(self.__warning) > 0:
+        if self.__warning:
             print "checking warning"
             self.__check_range('warning', name)
 
-        if len(self.__critical) > 0:
+        if self.__critical:
             print "checking critical"
             self.__check_range('critical', name)
 
